lib/src/git_backend.rs

```diff
@@ -93,7 +91,12 @@ impl From<GitBackendError> for BackendError {
 }
 
 pub struct GitBackend {
-    repo: Mutex<git2::Repository>,
+    // While gix::Repository can be created from gix::ThreadSafeRepository, it's
+    // cheaper to cache the thread-local instance behind a mutex than creating
+    // one for each backend method call. Our GitBackend is most likely to be
+    // used in a single-threaded context.
```

Comment:
Maybe `gix::ThreadSafeRepository` would be used in `ReadonlyGitBackend` and `gix::Repository` would be used in `MutableGitBackend` if we ever implement #50?

Reply:
I think both `ReadonlyGitBackend` and `MutableGitBackend` will use `gix::Repository`. Perhaps, `MutableGitBackend` wouldn't need `Mutex` as it wouldn't have to be `Sync`.

My understanding is that `gix::ThreadSafeRepository` is a pre-configured environment to instantiate a `gix::Repository` object (with some global cache?). It's something that a server preloads globally, and instantiate a repository object per user session for example.

Reply:
That's exactly the case, and the usage example is perfect for what `ThreadSafeRepository` and `Repository` are designed to do. This is also why `ThreadSafeRepository` can't do much more than spawn new `Repository` instances.